src/main/java/io/github/karmasmp/karmaplugin/KarmaPlayer.java

```diff
@@ -123,4 +141,22 @@ public void setVelocity(Vector vector) {
     public void teleport(Location location) {
         this.player.teleport(location);
     }
+
+    private void updateLives() {
+        if (this.lives > 0 && this.playerLifecycle.getCurrentPhase() instanceof SMPGhostPhase) {
```

Comment:
If a player starts with 3 lives and we decrement by 1, would this not redundantly tell them that they are now a player even though they already were one?

Also I think updatePlayerPhase or something would make more sense as updateLives does not actually do anything with lives other than read the value.

Reply:
> If a player starts with 3 lives and we decrement by 1, would this not redundantly tell them that they are now a player even though they already were one?

Nope, that's made sure by the `this.playerLifecycle.getCurrentPhase() instanceof SMPGhostPhase/SMPPlayerPhase` condition. By decreasing it to zero or lower while still being in a `SMPPlayerPhase` triggers the correct condition and applies necessary changes such as switching from `SMPPlayerPhase` to a `SMPGhostPhase`.

Reply:
Ok I see. What about?

> Also I think updatePlayerPhase or something would make more sense as updateLives does not actually do anything with lives other than read the value.

Reply:
> Also I think updatePlayerPhase or something would make more sense as updateLives does not actually do anything with lives other than read the value.

I guess that makes more sense.